docs/spec/fastsync/fastsync.md

```diff
@@ -0,0 +1,1164 @@
+     version
+	 of FastSync.
+	 
+   - [Suggestions](#Suggestions)  to address the issues discussed in the analysis.
+  
+
+In this document we quite extensively use tags in order to be able to
+reference assumptions, invariants, etc. in future communication. In
+these tags we frequently use the following short forms:
+
+- TMBC: Tendermint blockchain
+- SEQ: for sequential specifications
+- FS: Fastsync
+- LIVE: liveness
+- INV: invariant
+- TERM: termination
+- A: assumption
+- V2: refers to specifics of Fastsync V2
+- FS-VAR: refers to properties of Fastsync protocol variables
+- NewFS: refers to improved future Fastsync implementations
+
+# Part I - Outside view
+
+## Context of this document
+
+Fastsync is a protocol that is used by a full node to catch-up to the
+current state of a Tendermint blockchain. Its typical use case is a
+full node that was disconnected from the system for some time. The
+recovering full node locally has a copy of a prefix of the blockchain,
+and the corresponding application state that is slightly outdated. It
+then queries its peers for the blocks that were decided on by the
+Tendermint blockchain during the period the full node was
+disconnected. After receiving these blocks, it executes the
+transactions in the blocks in order to catch-up to the current height
+of the blockchain and the corresponding application state.
+
+In practice it is sufficient to catch-up only close to the current
+height: The Tendermint consensus reactor implements its own catch-up
+functionality and can synchronize a node that is close to the current height,
+perhaps within 10 blocks away from the current height of the blockchain.
+Fastsync should bring a node within this range.
+
+### Blockchain
+
+> We will briefly list some of the notions
+> of blockchains as required for this specification. More details can
+> be found  [here][block].
+
+#### **[TMBC-HEADER]**:
+A set of blockchain transactions is stored in a data structure called
+*block*, which contains a field called *header*. (The data structure
+*block* is defined [here][block]).  As the header contains hashes to
+the relevant fields of the block, for the purpose of this
+specification, we will assume that the blockchain is a list of
+headers, rather than a list of blocks. 
+
+#### **[TMBC-SEQ]**:
+
+The Tendermint blockchain is a list *chain* of headers. 
+
+#### **[TMBC-SEQ-GROW]**: 
+
+During operation, new headers may be appended to the list one by one.
+
+
+> In the following, *ETIME* is a lower bound
+> on the time interval between the times at which two
+> successor blocks are added. 
+
+#### **[TMBC-SEQ-APPEND-E]**: 
+If a header is appended at time *t* then no additional header will be
+appended before time *t + ETIME*.
+
+
+#### **[TMBC-AUTH-BYZ]**:
+The authenticated Byzantine model assumes that no node (faulty or
+correct) may break digital signatures, but otherwise, no additional
+assumption is made about the internal behavior of faulty full
```

Comment:
Full node is not defined at this point in time. Maybe it's sufficient to say here just faulty nodes.